assets/src/components/map/controls/LayersControl.tsx

```diff
@@ -0,0 +1,104 @@
+import Leaflet from "leaflet"
+import React, { useEffect, useState } from "react"
+import ReactDOM from "react-dom"
+import { TileLayer, useMap, useMapEvents } from "react-leaflet"
+import { joinClasses } from "../../../helpers/dom"
+import { TileType, tilesetUrlForType } from "../../../tilesetUrls"
+import { MapLayersIcon } from "../../../helpers/icon"
+
+export const LayersControl = ({
+  tileType,
+  setTileType,
+}: {
+  tileType: TileType
+  setTileType: (tileType: TileType) => void
+}): JSX.Element | null => {
+  const map = useMap()
+  const portalParent = map
+    .getContainer()
+    // TODO: handle two maps?
+    .querySelector(".leaflet-control-container .leaflet-top.leaflet-right")
+  const [portalElement, setPortalElement] = useState<HTMLElement | null>(null)
+
+  const [showLayersList, setShowLayersList] = useState(false)
+
+  useEffect(() => {
+    if (!portalParent || !portalElement) {
```

Comment:
**question:** is the `!portalParent` needed? It looks like this means that `portalParent` could be null but `portalElement` defined and then it would cause us to create a new `portalElement`, only to repeat again whenever `portalParent` becomes defined again.

Aside from that though I _think_ the logic in this hook is sound.

Reply:
(Though I do realize that this is how I myself did it in #1834, so maybe I had my reasons?)